src/cmake/externalpackages.cmake

```diff
@@ -147,8 +147,10 @@ if (LLVM_VERSION VERSION_GREATER_EQUAL 15.0
          "If you are using LLVM 15 or higher, you should also use clang version "
          "15 or higher, or you may get build errors.${ColorReset}\n")
 endif ()
+if (LLVM_VERSION VERSION_GREATER_EQUAL 18.0)
+    message (ERROR "${ColorYellow}OSL is not yet compatible with LLVM 18.${ColorReset}\n")
+endif ()
```

Comment:
I'm not sure these lines are necessary. The `VERSION_MAX 17.9` when searching for LLVM (and the `REQUIRED` marker) will already error out about not having the right LLVM version before reaching this point.